src/components/sidebar/PopoeverComponent.js

```diff
@@ -0,0 +1,65 @@
+import AddIcon from '@mui/icons-material/Add';
+import { Popover, ListItemButton, ListItemText } from '@mui/material';
+import { useState } from 'react';
+import './Sidebar.css';
+
+export default function PopoverComponent({ services }) {
+  const [popoverOpen, setPopovereOpen] = useState(null);
+  const [serviceFormOpen, setServiceFormOpen] = useState(false);
+
+  const addIconOnclick = (e) => {
+    e.stopPropagation();
+    setPopovereOpen(e.target);
+  };
+
+  const handlePopoever = (e) => {
+    e.stopPropagation();
+    setPopovereOpen(null);
+  };
+
+  const handleService = (service, e) => {
```

Comment:
Can we move this implementation somewhere else? Otherwise, they are still strongly connected with each other.